python/vision_explanation_methods/explanation_evaluation/pointing_game.py

```diff
@@ -0,0 +1,179 @@
+
+        img_input = (T.ToTensor()(test_image)
+                     .unsqueeze(0).to(self._device))
+
+        detections = self._model.predict(img_input)
+
+        saliency_scores = drise.DRISE_saliency(
+                model=self._model,
+                # Repeated the tensor to test batching
+                image_tensor=img_input,
+                target_detections=detections,
+                # This is how many masks to run -
+                # more is slower but gives higher quality mask.
+                number_of_masks=num_masks,
+                mask_padding=None,
+                device=self._device,
+                # This is the resolution of the random masks.
+                # High resolutions will give finer masks, but more need to be
+                # run.
+                mask_res=(2, 2),
+                verbose=True  # Turns progress bar on/off.
+            )
+
+        temp = saliency_scores[0][index]['detection']
+
+        temp[temp < threshold] = -1
+        return temp
+
+    def visualize_highly_salient_pixels(self,
+                                        img,
+                                        saliency_scores):
+        """
+        Create figure of highly salient pixels.
+
+        :param imagelocation: Path of the image location
+        :type imagelocation: str
+        :param saliency_scores: 2D matrix representing the saliency scores
+            of each pixel in an image
+        :type saliency_scores: List[Tensor]
+        :return: Overlay of the saliency scores on top of the image
+        :rtype: Figure
+        """
+        fig, ax = pl.subplots(1, 1, figsize=(10, 10))
+
+        viz.visualize_image_attr(
+            np.transpose(
+                saliency_scores.cpu().detach().numpy(),
+                (1, 2, 0)),
+            np.transpose(T.ToTensor()(img).numpy(), (1, 2, 0)),
+            method="blended_heat_map",
+            sign="positive",
+            show_colorbar=True,
+            cmap=pl.cm.get_cmap("Blues"),
+            title="Pointing Game Visualization",
+            plt_fig_axis=(fig, ax),
+            use_pyplot=False
+        )
+
+        return fig
+
+    def calculate_gt_salient_pixel_overlap(self,
+                                           saliency_scores: List[Tensor],
+                                           gt_bbox: List):
+        """
+        Calculate percent of overlap between salient pixels and gt bbox.
+
+        :param saliency_scores: 2D matrix representing the saliency scores
+            of each pixel in an image
+        :type saliency_scores: List[Tensor]
+        :param gt_bbox: bounding box for ground truth prediction
+        :type gt_bbox: List
+        :return: return percent of salient pixel overlap with the ground truth
+        :rtype: Float
+        """
+        good = 0
+        total = 0
+        total_percent = 0
+        for iindex, i in enumerate(saliency_scores):
+            for jindex, j in enumerate(i):
```

Comment:
For the for loop starting at 149, where `i` is a tensor, I'm curious if `good` & `total` can be calculated with torch operations.

Something like
```python
good += torch_operations(i)
total += torch_operations2(i)
```

Curious if it may be faster due to torch's optimizations (even though you're using a single for loop here and the torch operation approach may be going through each tensor twice).

Could explore if have bandwidth on how many such math operations / metrics computations can be offloaded to torch. Happy to assist as needed.